### DISCUSSION TITLE
Dynamic Q-Select with v-model returns do not mutate vuex store state error

### DISCUSSION THREAD

**Discussion by ezanglo** [quasarframework/quasar#9990] [General - Components / Directives / etc]
Hi! I'm pretty new with Quasar and vue, I was trying to generate dynamic q-select, assign a v-model and update it. 
I have an array of values which creates the q-select, the problem is when i'm trying to update it, it gives me a do not mutate vuex store state error. I can't find a way to figure this out since it's directly attached to the v-model, i also try doing a clone of the array by using [...array] and updating that instead, but still no luck. 
Here is a snippet 
```
<div v-for="(inventoryItem, index) in currentInventoryKit.inventoryKitItems" :key="index" class="q-gutter-md row">
  <q-select label="Inventory Item" class="col-7" stack-label lazy-rules="lazy-rules" map-options emit-value
            :options="inventoryItemOptions"
            v-model="currentInventoryKitItems.unique_id
            :rules="[val => !!val || '*Field is required']"/>
  <q-input label="Quantity" class="col-2" stack-label lazy-rules="lazy-rules" type="number"
           v-model="inventoryItem.quantity" :rules="[val => !!val || '*Field is required']"/>
  <q-btn round flat icon="delete" @click="removeInventoryItem(index)"/>
</div>
```

*Update
I was able to fix this by making sure `inventoryItem` object has the exact same fields as the `inventoryItemOptions`, i think the mutation error occurs when it tried to remove the other fields from `inventoryItem` when changing the values of the q-select.

**Comment by metalsadman** [comment]
@ezanglo please refer to vuex docs, like the error suggests, you are directly mutating a store state. you can either clone your state into a local component data or implement proper vuex mutation of your state.

**Reply by ezanglo** [reply]
Hi! Thank you for the response! I was actually able to fix this by making sure that the object inside the said array have the same exact fields as the options, it looks like it's trying to remove the other ones when i'm trying to set it thats why the mutate error.
Thank you!

**Comment by ezanglo** [comment]
*Update
I was able to fix this by making sure `inventoryItem` object has the exact same fields as the `inventoryItemOptions`, i think the mutation error occurs when it tried to remove the other fields from `inventoryItem` when changing the values of the q-select.
